### PR TITLE
Update the k8s card and registration flow based on feedback

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
@kenxu95 I think we need to rebase backend-eng-2864 with `main` first.
Loom demo here: https://www.loom.com/share/ba510a359b0141579dd5acde5333ee5c

Also adds the ECR resource details card.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/integrations/addIntegrations.tsx [hunk 1]
```diff
@@ -148,20 +162,231 @@ const AddIntegrationListItem: React.FC<AddIntegrationListItemProps> = ({
     </Box>
   );
 
+  if (service !== 'Kubernetes') {
+    return (
+      <Box key={service}>
```
Comment (agiron123) [inline src/ui/common/src/components/integrations/addIntegrations.tsx:167]: Can you abstract this out into it's own component please?

File: src/ui/common/src/components/integrations/addIntegrations.tsx [hunk 2]
```diff
@@ -148,20 +162,231 @@ const AddIntegrationListItem: React.FC<AddIntegrationListItemProps> = ({
...
+    setShowOndemandDialog(true);
+    setShowSelectProviderDialog(false);
+  };
+
   return (
     <Box key={service}>
       <Box>
         {iconWrapper}
-        {showDialog && (
+        <Dialog open={showDialog} onClose={() => setShowDialog(false)}>
```
Comment (agiron123) [inline src/ui/common/src/components/integrations/addIntegrations.tsx:229]: Same here, this is very Kubernetes specific and has nothing to do with the other integrations. This should have it's own file somewhere.

File: src/ui/common/src/components/integrations/cards/ecrCard.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    values.push(config.config_file_path);
+  }
+
+  if (config.config_file_profile) {
+    labels.push('Profile');
+    values.push(config.config_file_profile);
+  }
+
+  return (
+    <Box>
```
Comment (agiron123) [inline src/ui/common/src/components/integrations/cards/ecrCard.tsx:33]: This box isn't really needed anymore. you can just return the ResourceCardText component.

File: src/ui/common/src/components/integrations/connectedIntegrations.tsx [hunk 4]
```diff
@@ -111,6 +111,10 @@ export const ConnectedIntegrations: React.FC<ConnectedIntegrationsProps> = ({
         }}
       >
         {[...integrations]
+          .filter(
```
Comment (agiron123) [inline src/ui/common/src/components/integrations/connectedIntegrations.tsx:114]: Would you mind leaving a comment here explaining that this is a bit of a temporary fix?

**Review by agiron123 — APPROVED** [review]
Nice work. I've left a couple small comments here that you can resolve, or file a Linear task for and get to after this coming release.

There are a couple things that will conflict my forms refactoring work, but I should be able to work around them.

**Comment by agiron123** [comment]
It's also worth noting that this PR contains a fix for ENG-2887, which redirects users to the Resources list page after deleting a resource, instead of the old "Integrations" page.

File: src/ui/common/src/components/integrations/addIntegrations.tsx (hunk 1 above)
Comment (cw75) [reply src/ui/common/src/components/integrations/addIntegrations.tsx:167]: Yeah I'm tracking it in another Linear task: https://linear.app/aqueducthq/issue/ENG-2876/refactor-on-demand-k8s-registration-flow-into-a-separate-component

Comment (agiron123) [reply src/ui/common/src/components/integrations/addIntegrations.tsx:167]: Great. Thanks for filing a task for this.

**Pull request merged**